inference-engine/src/mkldnn_plugin/nodes/mkldnn_deconv_node.h

```diff
@@ -39,27 +43,120 @@ class MKLDNNDeconvolutionNode : public MKLDNNNode {
     static bool isSupportedOperation(const std::shared_ptr<const ngraph::Node>& op, std::string& errorMessage) noexcept;
     bool canFuse(const MKLDNNNodePtr& node) const override;
 
-    const InferenceEngine::SizeVector& getWeightDims() { return weightDims; }
-    const std::vector<ptrdiff_t>& getStride() { return stride; }
+    const VectorDims& getWeightDims() const { return getInputShapeAtPort(1).getStaticDims(); }
+    const std::vector<ptrdiff_t>& getStride() const { return stride; }
+
+    void prepareParams() override;
+    void execute(mkldnn::stream strm) override;
+    void executeDynamicImpl(mkldnn::stream strm) override { execute(strm); }
+    bool needShapeInfer() const override;
+    std::vector<VectorDims> shapeInfer() const override;
 
 private:
+    class DeconvExecutor {
+        protected:
+            class IntermReorder {
+                public:
+                    IntermReorder(MKLDNNMemoryPtr memFrom, const mkldnn::memory::desc& descTo, const mkldnn::engine& engine);
+                    IntermReorder(const mkldnn::memory::desc& descFrom, MKLDNNMemoryPtr memTo, const mkldnn::engine& engine);
+                    MKLDNNMemoryPtr getFromMem() const { return m_memFrom; }
+                    MKLDNNMemoryPtr getToMem() const { return m_memTo; }
+                    void exec(mkldnn::stream strm);
+
+                private:
+                    MKLDNNMemoryPtr m_memFrom;
+                    MKLDNNMemoryPtr m_memTo;
+                    mkldnn::reorder m_reorder;
+            };
+
+        public:
+            void exec(mkldnn::stream strm);
+            virtual ~DeconvExecutor() = default;
+
+        protected:
+            DeconvExecutor() = default;
+            std::vector<IntermReorder> inputReorders;
+            MKLDNNPrimitive execPrim;
+            std::vector<IntermReorder> outputReorders;
+            std::unordered_map<int, mkldnn::memory> primArgs;
```

Comment:
IMPORTANT. This is runtime paramters and MUST NOT be a part of the executor, since they are pointers to concrete memory, but the executor can be shared across nodes. This parameters must be passed to the exec method at the executon stage.

Reply:
At this moment we don't enable primitive caching it will work.
So if I got green CI is it possible to merge it as is? and I will fix it in https://github.com/openvinotoolkit/openvino/pull/8871

Reply:
@mandrono @maxnick my recommendation is to fix in follow-up PR.